lib/Onyx.js

```diff
@@ -610,4 +618,25 @@ const Onyx = {
     isSafeEvictionKey,
 };
 
+function applyDecorators() {
+    // We're requiring the script dynamically here so that it's only evaluated when decorators are used
+    const decorate = require('./decorateWithMetrics');
+
+    // Re-assign with decorated functions
+    /* eslint-disable no-func-assign */
+    get = decorate.decorateWithMetrics(get, 'Onyx:get');
+    set = decorate.decorateWithMetrics(set, 'Onyx:set');
+    multiSet = decorate.decorateWithMetrics(get, 'Onyx:multiSet');
+    clear = decorate.decorateWithMetrics(get, 'Onyx:clear');
+    merge = decorate.decorateWithMetrics(get, 'Onyx:merge');
+    mergeCollection = decorate.decorateWithMetrics(get, 'Onyx:mergeCollection');
+    AsyncStorage.getAllKeys = decorate.decorateWithMetrics(AsyncStorage.getAllKeys, 'AsyncStorage:getAllKeys');
+    /* eslint-enable */
+
+    // Expose stats methods on Onyx
+    Onyx.getMetrics = decorate.getMetrics;
+    Onyx.resetMetrics = decorate.resetMetrics;
+    Onyx.isCapturingMetrics = true;
```

Comment:
What does `isCapturingMetrics` do for us? If we have set `captureMetrics` to `true` then we should know that we are capturing metrics?

Reply:
* Not a blocker

Reply:
>This way we don't have to import and test for the ENV variable that sets this flag again

seems reasonable to use an env variable

Reply:
This is a NAB in any case, but perhaps we should initialize Onyx so that `Onyx.isCapturingMetrics` returns `false`.

Reply:
> > This way we don't have to import and test for the ENV variable that sets this flag again
> 
> seems reasonable to use an env variable

It's not very easy to find out during debugging - you can't check `process.env.ONYX_BENCHMARK` - it's not set there
It's set at the end of the decoration - you know it passed successfully when this is set to `true` 



> This is a NAB in any case, but perhaps we should initialize Onyx so that `Onyx.isCapturingMetrics` returns `false`.

I don't know what you mean by that, we do initialize to to `false` by default: `captureMetrics = false,`
It's only `true` when we want benchmarking

Reply:
I'm suggesting we explicitly set the value to `false`, but again, it's not a blocker. I get that this `Onyx.isCapturingMetrics` will return `undefined` which is falsy, but maybe more clear if it was always a bool.

Reply:
>It's not very easy to find out during debugging

Again not a blocker, but we import env variables to the `Config` object so it is very easy (at least for us) to find out 😂